Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/PDFDiscretePresentationController.mm

```diff
@@ -1069,19 +1079,20 @@ static float elasticDeltaForTimeDelta(float initialPosition, float initialVeloci
 
         auto rightPageIndex = layoutRow.pages[1];
         row.rightPageContainerLayer = makePageContainerLayer(rightPageIndex);
-        RefPtr rightPageBackgroundLayer = pageBackgroundLayerForPageContainerLayer(*row.rightPageContainerLayer);
+        Ref rowRightPageContainerLayer = *row.rightPageContainerLayer.get();
+        RefPtr rightPageBackgroundLayer = pageBackgroundLayerForPageContainerLayer(rowRightPageContainerLayer.get());
         m_layerIDToRowIndexMap.add(*rightPageBackgroundLayer->primaryLayerID(), rowIndex);
     };
 
     auto parentRowLayers = [](RowData& row) {
         ASSERT(row.containerLayer->children().isEmpty());
-        row.containerLayer->addChild(*row.leftPageContainerLayer);
+        row.containerLayer->addChild(row.leftPageContainerLayer.releaseNonNull());
```

Comment:
This will clear `row.leftPageContainerLayer`. Is that what we want to do here?

Reply:
This looks like a potential bug to me.

Reply:
@basuke the `releaseNonNull()` calls in this lambda will cause issues because we don't want to clear the `[left|right]PageContainerLayer` members.

You can safely keep dereferencing here.

Reply:
Got it. Sorry I read the code wrong.